arangod/Cluster/ClusterFeature.cpp

```diff
@@ -420,6 +420,22 @@ any extra JWT checks compared to v3.7.)");
 shards operations that can be made when the **Rebalance Shards** button is
 clicked in the web interface. For backwards compatibility, the default value is
 `10`. A value of `0` disables the button.)");
+
+  options
+      ->addOption("--cluster.failed-write-concern-error-code",
+                  "HTTP code which is sent when a shard has not enough in-sync "
+                  "replicas to fulfill the write concern.",
+                  new UInt32Parameter(&_returnCodeFailedWriteConcern),
+                  arangodb::options::makeFlags(
+                      arangodb::options::Flags::DefaultNoComponents,
+                      arangodb::options::Flags::OnDBServer))
+      .setIntroducedIn(31100)
+      .setLongDescription(
+          R"(The old legacy behavior is to send a 403 FORBIDDEN. The new,
```

Comment:
Nit: "old legacy" seems redundant.

Reply:
I'd argue that it's not legacy at all because it is the default.

Reply:
My thinking was that it is plain "wrong" to answer "403 FORBIDDEN collection is read-only". First of all, why is it forbidden to write, and secondly, it is a temporary error and the operation is retryable, so it should be 503 and should say what the actual problem is. Therefore "legacy" and "correct".